appveyor/numpy.patch

```diff
@@ -0,0 +1,831 @@
+ from distutils.errors import DistutilsFileError, DistutilsSetupError,\
+-     DistutilsError
++    DistutilsError
+ from distutils.file_util import copy_file
+ 
+ from numpy.distutils import log
+ from numpy.distutils.exec_command import exec_command
+ from numpy.distutils.system_info import combine_paths
+ from numpy.distutils.misc_util import filter_sources, has_f_sources, \
+-     has_cxx_sources, get_ext_source_files, \
+-     get_numpy_include_dirs, is_sequence, get_build_architecture, \
+-     msvc_version
++    has_cxx_sources, get_ext_source_files, \
++    get_numpy_include_dirs, is_sequence, get_build_architecture, \
++    msvc_version
+ from numpy.distutils.command.config_compiler import show_fortran_compilers
+ 
+ try:
+@@ -27,6 +30,7 @@ try:
+ except NameError:
+     from sets import Set as set
+ 
++
+ class build_ext (old_build_ext):
+ 
+     description = "build C/C++/F extensions (compile/link to build directory)"
+@@ -36,17 +40,18 @@ class build_ext (old_build_ext):
+          "specify the Fortran compiler type"),
+         ('parallel=', 'j',
+          "number of parallel jobs"),
+-        ]
++    ]
+ 
+     help_options = old_build_ext.help_options + [
+         ('help-fcompiler', None, "list available Fortran compilers",
+          show_fortran_compilers),
+-        ]
++    ]
+ 
+     def initialize_options(self):
+         old_build_ext.initialize_options(self)
+         self.fcompiler = None
+         self.parallel = None
++        self.dlls = []
+ 
+     def finalize_options(self):
+         if self.parallel:
+@@ -84,11 +89,13 @@ class build_ext (old_build_ext):
+         if self.distribution.has_c_libraries():
+             if self.inplace:
+                 if self.distribution.have_run.get('build_clib'):
+-                    log.warn('build_clib already run, it is too late to ' \
+-                            'ensure in-place build of build_clib')
+-                    build_clib = self.distribution.get_command_obj('build_clib')
++                    log.warn('build_clib already run, it is too late to '
++                             'ensure in-place build of build_clib')
++                    build_clib = self.distribution.get_command_obj(
++                        'build_clib')
+                 else:
+-                    build_clib = self.distribution.get_command_obj('build_clib')
++                    build_clib = self.distribution.get_command_obj(
++                        'build_clib')
+                     build_clib.inplace = 1
+                     build_clib.ensure_finalized()
+                     build_clib.run()
+@@ -98,6 +105,8 @@ class build_ext (old_build_ext):
+                 self.run_command('build_clib')
+                 build_clib = self.get_finalized_command('build_clib')
+             self.library_dirs.append(build_clib.build_clib)
++            self.dlls.extend(build_clib.dlls)
++            self.build_clib = build_clib.build_clib
```

Comment:
We need to be able to link fortran code built here to potentially any DLL built in build_clib.